ambry-router/src/main/java/com.github.ambry.router/PutOperation.java

```diff
@@ -1211,22 +1225,43 @@ void onFillComplete(boolean updateMetric) {
     }
 
     /**
-     * Fill the buffer of the current chunk with the data from the given {@link ByteBuffer}.
-     * @param channelReadBuffer the {@link ByteBuffer} from which to read data.
+     * Fill the buffer of the current chunk with the data from the given {@link ByteBuf}.
+     * @param channelReadBuf the {@link ByteBuf} from which to read data.
      * @return the number of bytes transferred in this operation.
      */
-    int fillFrom(ByteBuffer channelReadBuffer) {
-      int toWrite = Math.min(channelReadBuffer.remaining(), buf.remaining());
-      if (channelReadBuffer.remaining() > buf.remaining()) {
-        // Manipulate limit of the source buffer in order to read only enough to fill the chunk
-        int savedLimit = channelReadBuffer.limit();
-        channelReadBuffer.limit(channelReadBuffer.position() + buf.remaining());
-        buf.put(channelReadBuffer);
-        channelReadBuffer.limit(savedLimit);
+    int fillFrom(ByteBuf channelReadBuf) {
+      int toWrite = 0;
```

Comment:
for declaring`toWrite` and `remaininSlice` you can remove the assignment part because of each of them are assigned by all the cases afterwards

Reply:
updated to new code.